### ISSUE TITLE
Update .gov content guide linking recommendations to align with accessibility best practices

### ISSUE THREAD

**Issue by rileyorr** [cisagov/manage.get.gov#2107]
### Issue description

Some of the links on get.gov open in a new tab while others don't. W3C recommends opening new windows and tabs from a link _only_ when necessary since it can be disorienting for users and difficult to navigate back to the previous page ([source](https://www.w3.org/TR/WCAG20-TECHS/G200.html)).

We should reevaluate our linking strategy as a team and update [the .gov content guide](https://docs.google.com/document/d/1U-TRx3ecCFZ-qtrCk7EYmF_nmfKh6kd7FV95Pa4iWeM/edit#heading=h.uu1tgvdl81ni) as appropriate. 

### Acceptance criteria

- [x] W3C's recommendation has been shared with the team (ex. in product + design huddle) for alignment on guidelines related to opening links in a new tab vs not 
- [x] A recommendation based on team feedback has been documented and shared with Katherine and Cameron for approval 
- [x] The .gov content style guide is updated as needed and shared with the team 

### Links to other issues

Relates to #1917 
Additional context can be found in https://github.com/cisagov/get.gov/pull/280 

### Prioritization Comments
**WCAG Conformance Level:** 2.1 AA
**WCAG Standard that applies:** 
3.2.5 (Change on Request)
**Critical Failure if Not Met?:**  No
**Prioritization:** Normal/Should Do
**Reason:**

**Comment by AnnaGingle** [comment]
cc @joet333 Kinda related to the inconsistent link patterns.

**Comment by SamiyahKey** [comment]
The following policy for jump-links was added to the content guide and can be seen [[here](https://docs.google.com/document/d/1U-TRx3ecCFZ-qtrCk7EYmF_nmfKh6kd7FV95Pa4iWeM/edit?tab=t.0#heading=h.x7btrba3h2cw)] :

### Jump-links (in-page links)
Jump links (or in-page links) will navigate the user to another part of the same page. 

In general, this behavior can be unexpected in header and body text links. In these cases, use the link text like “jump to”, “above” or “below” to tell users that the destination is elsewhere on the same page.
Example text in a jump-link
[Jump to video resources](https://docs.google.com/document/d/1U-TRx3ecCFZ-qtrCk7EYmF_nmfKh6kd7FV95Pa4iWeM/edit?tab=t.0#heading=h.x7btrba3h2cw) for more information about how to start a domain request.
